site/docs/v1/tech/migration-guide/firebase.adoc

```diff
@@ -0,0 +1,390 @@
+
+After installing the CLI, be sure to
+https://firebase.google.com/docs/cli#sign-in-test-cli[login and test]
+that the installation is successful.
+
+=== Retrieve the Project ID
+
+Before exporting the users, you'll need the Project ID of the Firebase
+project you want to export from. To get this ID, run the following
+Firebase CLI command:
+
+[source,bash]
+----
+firebase projects:list
+----
+
+This will list all your Firebase projects, like this:
+
+....
+│ Project Display Name │ Project ID   │ Project Number │ Resource Location ID │
+├──────────────────────┼──────────────┼────────────────┼──────────────────────┤
+│ fusion               │ fusion-bea44 │ 1021914842301  │ [Not specified]   
+....
+
+Make a note of the Project ID of the Firebase project you want to export
+from.
+
+=== Download the File
+
+To download the user file, run the following
+https://firebase.google.com/docs/cli/auth[Firebase `auth:export` CLI
+command]:
+
+[source,bash]
+----
+firebase auth:export users.json --format=JSON --project your_project_id
+----
+
+Replace `your_project_id` with the Project ID you noted above.
+
+After the export finishes, you’ll end up with a JSON file called
+`users.json`. It should look something like this:
+
+[source,js]
+----
+{"users": [
+    {
+      "localId": "OzDdXA7LwoR7lX2MH7AXaEmmn5u2",
+      "email": "user1@test.com",
+      "emailVerified": false,
+      "passwordHash": "0fn2PA6FmYZynpk9cvekSgbJTXa7j0XQAwtp4XuyyuIYzX5hASd4mB4GFeaS5OiG9mENrvt+sPoZmwVjvEDZ2Q==",
+      "salt": "+mkMRRbwdwqJkA==",
+      "createdAt": "1648020042135",
+      "disabled": false,
+      "providerUserInfo": []
+    },
+    {
+      "localId": "bBd018SFAYa8fkkZdgAz3PEgaKj1",
+      "email": "user2@test.com",
+      "emailVerified": false,
+      "passwordHash": "TFJtUjKMN4dNcp5IuSwaeRkPwjwpkp9ZlqXuL/QHsQ3097QLHnZWccWt2yThLa0Q5rmbuOOXoqzoHBZM8x3GpQ==",
+      "salt": "RHQ5jbaxNJ1lDA==",
+      "createdAt": "1648020072141",
+      "disabled": false,
+      "providerUserInfo": []
+    }
+]}
+----
+
+== Importing Users
+
+Next up, import the user data. Here are the steps we need to take.
+
+1. Set Up FusionAuth
+2. Add the scrypt Hashing Plugin
+3. Get the Script
+4. Install Needed Gems
+5. Use the Script
+6. Verify the Import
+7. The Final Destination of Imported Users
+
+=== Set Up FusionAuth
+
+include::docs/v1/tech/migration-guide/_set-up-fusionauth.adoc[]
+
+==== Create a Test Tenant
+
+include::docs/v1/tech/migration-guide/_create-test-tenant.adoc[]
+
+==== Create a Test Application
+
+include::docs/v1/tech/migration-guide/_create-test-application.adoc[]
+
+==== Add an API Key
+
+include::docs/v1/tech/migration-guide/_create-api-key.adoc[]
+
+=== Add the scrypt Hashing Plugin
+
+Most Firebase projects use a https://github.com/firebase/scrypt[modified
+version of the scrypt algorithm] for password hashes. FusionAuth does
+not support scrypt directly, but does
+https://fusionauth.io/docs/v1/tech/plugins/custom-password-hashing[support
```

Comment:
```suggestion
link:/docs/v1/tech/plugins/custom-password-hashing[support
```

We should keep all links to FusionAuth relative so that we can build this locally and click around without being sent off to prod.